core/domain/opportunity_services.py

```diff
@@ -0,0 +1,380 @@
+    """Returns the ExplorationOpportunitySummary object out of the model.
+
+    Args:
+        model: ExplorationOpportunitySummaryModel. The exploration opportunity
+            summary model.
+
+    Returns:
+        ExplorationOpportunitySummary. The corresponding
+        ExplorationOpportunitySummary object.
+    """
+
+    return opportunity_domain.ExplorationOpportunitySummary(
+        model.id, model.topic_id, model.topic_name, model.story_id,
+        model.story_title, model.chapter_title, model.content_count,
+        model.incomplete_translation_languages, model.translation_counts,
+        model.need_voice_artist_in_languages,
+        model.assigned_voice_artist_in_languages)
+
+
+def _save_multi_exploration_opportunity_summary(
+        exploration_opportunity_summary_list):
+    """Stores multiple ExplorationOpportunitySummary into datastore as a
+    ExplorationOpportunitySummaryModel.
+
+    Args:
+        exploration_opportunity_summary_list: list(
+            ExplorationOpportunitySummary). A list of exploration opportunity
+            summary object.
+    """
+    exploration_opportunity_summary_model_list = []
+    for opportunity_summary in exploration_opportunity_summary_list:
+        model = opportunity_models.ExplorationOpportunitySummaryModel(
+            id=opportunity_summary.id,
+            topic_id=opportunity_summary.topic_id,
+            topic_name=opportunity_summary.topic_name,
+            story_id=opportunity_summary.story_id,
+            story_title=opportunity_summary.story_title,
+            chapter_title=opportunity_summary.chapter_title,
+            content_count=opportunity_summary.content_count,
+            incomplete_translation_languages=(
+                opportunity_summary.incomplete_translation_languages),
+            translation_counts=opportunity_summary.translation_counts,
+            need_voice_artist_in_languages=(
+                opportunity_summary.need_voice_artist_in_languages),
+            assigned_voice_artist_in_languages=(
+                opportunity_summary.assigned_voice_artist_in_languages)
+        )
+
+        exploration_opportunity_summary_model_list.append(model)
+
+    opportunity_models.ExplorationOpportunitySummaryModel.put_multi(
+        exploration_opportunity_summary_model_list)
+
+
+def add_new_exploration_opportunities(story, exp_ids):
+    """Adds new exploration opportunity into the model.
+
+    Args:
+        story: Story. A story object.
+        exp_ids: list(str). A list of exploration ids for which new
+            opportunities are to be created.
+    """
+    topic = topic_services.get_topic_by_id(story.corresponding_topic_id)
+    explorations = exp_fetchers.get_multiple_explorations_by_id(exp_ids)
+
+    exploration_opportunity_summary_list = []
+    for exp_id, exploration in explorations.iteritems():
+        node = story.story_contents.get_node_with_corresponding_exp_id(exp_id)
```

Comment:
What if the exp_id isn't in the story? Raise exception? Or skip?

Probably former is better, but then make sure that callers know this, and include a stipulation in the docstring after the exp_ids description: "All exp_ids must be part of the given story."

Reply:
Still validate, please. Other functions may call this one in the future, so better to be safe.